strongbox-storage/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/services/impl/ArtifactManagementServiceImpl.java

```diff
@@ -133,24 +141,54 @@ public void store(String storageId,
         }
     }
 
-    private void addArtifactToIndex(String storageId, String repositoryId, String path)
-            throws IOException
+    private void storeArtifact(String storageId,
+                               String repositoryId,
+                               String path)
     {
-        if (ArtifactFileUtils.isArtifactFile(path))
-        {
-            final RepositoryIndexer indexer = repositoryIndexManager.getRepositoryIndexer(storageId + ":" + repositoryId);
-            if (indexer != null)
-            {
-                final Artifact artifact = ArtifactUtils.convertPathToArtifact(path);
-                final Storage storage = getStorage(storageId);
-                final File storageBasedir = new File(storage.getBasedir());
-                final File artifactFile = new File(new File(storageBasedir, repositoryId), path).getCanonicalFile();
+        ArtifactCoordinates artifactCoordinates = artifactResolutionService.getArtifactCoordinates(storageId,
+                                                                                                   repositoryId, path);
 
-                if (!artifactFile.getName().endsWith(".pom"))
-                {
-                    indexer.addArtifactToIndex(repositoryId, artifactFile, artifact);
-                }
-            }
+        ArtifactEntry artifactEntry = artifactEntryService.findOne(artifactCoordinates)
+                                                          .orElse(createArtifactEntry(artifactCoordinates, storageId,
+                                                                                      repositoryId));
+        artifactEntryService.save(artifactEntry);
+    }
+
+    private ArtifactEntry createArtifactEntry(ArtifactCoordinates artifactCoordinates,
+                                              String storageId,
+                                              String repositoryId)
+    {
+        ArtifactEntry artifactEntry = new ArtifactEntry();
+        artifactEntry.setStorageId(storageId);
+        artifactEntry.setRepositoryId(repositoryId);
+        artifactEntry.setArtifactCoordinates(artifactCoordinates);
+        return artifactEntry;
+    }
+
+    private void addArtifactToIndex(String storageId,
+                                    String repositoryId,
+                                    String path)
+        throws IOException
+    {
+        if (!ArtifactFileUtils.isArtifactFile(path))
+        {
+            return;
+        }
+        
+        RepositoryIndexer indexer = repositoryIndexManager.getRepositoryIndexer(storageId + ":" + repositoryId);
```

Comment:
@sbespalov : I have the feeling that something here has not been merged properly. The `RepositoryIndexManager`'s `contextId`-s now are formed of three components - `storageId`, `repositoryId`, `indexType`. Here are valid examples:
- `storage0:releases:local`
- `storage0:releases:remote`

This might be the reason you have issues with the search results.